xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -350,20 +467,81 @@ viewers.Comments = Class.create({
       this.initialLocation.insert({after: this.form});
     }
     this.form["XWiki.XWikiComments_replyto"].value = "";
-    this.form["XWiki.XWikiComments_comment"].value = "";
+    this.reloadEditor();
     this.cancelPreview(this.form);
   },
   /**
    * Registers a listener that watches for the insertion of the Comments tab and triggers the enhancements.
    * After that, the listener removes itself, since it is no longer needed.
    */
-  addTabLoadListener : function(event) {
-    var listener = function(event) {
+  addTabLoadListener: function (event) {
+    var listener = function (event) {
       if (event.memo.id == 'Comments') {
         this.startup();
       }
     }.bindAsEventListener(this);
     document.observe("xwiki:docextra:loaded", listener);
+  },
+  destroyEditor: function (selector, name) {
+    require(['jquery', 'xwiki-events-bridge'], function ($) {
+      $(document).trigger('xwiki:actions:cancel');
+      if ($(selector).size() > 0) {
+        const parent = $(selector).parent(".commenteditor");
+        parent.empty();
+        $(document).trigger('xwiki:dom:updated', {'elements': parent.toArray()});
```

Comment:
I don't see the point of triggering the ``'xwiki:dom:updated'`` for an element that was just emptied. ``'xwiki:dom:updated'`` is useful when you add or change the DOM, because it allows others to enhance the added DOM. But here a listener that is called for an empty element won't be able to do much. And doing cleanup is not a valid reason because the listener won't know what content was removed (it just receives the empty parent). I would simply drop this line.